exercises/binary-search/example.exs

```diff
@@ -0,0 +1,35 @@
+defmodule BinarySearch do
+  @doc """
+    Searches for a key in the list using the binary search algorithm.
+    It returns :not_found if the key is not in the list.
+    Otherwise returns the tuple {:ok, index}.
+
+    ## Examples
+
+      iex> BinarySearch.search([], 2)
+      :not_found
+
+      iex> BinarySearch.search([1, 3, 5], 2)
+      :not_found
+
+      iex> BinarySearch.search([1, 3, 5], 5)
+      {:ok, 2}
+
+  """
+
+  @spec search(Enumerable.t, integer) :: {:ok, integer} | :not_found
+  def search([], _key), do: :not_found
+  def search(list, key), do: _search(list, key, 0, length(list) - 1)
+
+  defp _search(list, _key, low, high) when high < low, do: :not_found
+  defp _search(list, key, low, high) do
```

Comment:
you'll see it a lot if you read some of the elixir source, especially in the `Enum` module if you want to take a look: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/enum.ex